enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -576,12 +581,16 @@ func (p *PebbleCache) scanForBrokenFiles(quitChan chan struct{}) error {
 	fileMetadata := &rfpb.FileMetadata{}
 	blobDir := ""
 
+	defer func() {
+		close(p.brokenFilesDone)
+	}()
+
 	mismatchCount := 0
 	for iter.Next() {
 		// Check if we're shutting down; exit if so.
 		select {
 		case <-quitChan:
-			break
+			return nil
```

Comment:
this was a bug before -- previously break was only exiting the select loop and this goroutine was not stopping